core/net/mac/tsch/sixtop/README.md

```diff
@@ -0,0 +1,94 @@
+# [EXPERIMENTAL] 6TiSCH Operation Sublayer (6top)
+
+## Overview
+
+6TiSCH Operation Sublayer (6top), a logical link layer in the 6TiSCH
+architecture, provides the abstraction of an IP link over a TSCH MAC layer (IEEE
+802.15.4e). The functionality of 6top is to facilitate the dynamic cell
+negotiation with one-hop neighbors.
+
+This is a Contiki implementation of 6top, developed by:
+
+- Shalu R, CDAC, shalur@cdac.in, github user: [shalurajendran](https://github.com/shalurajendran)
+- Lijo Thomas, CDAC, lijo@cdac.in
+- Yasuyuki Tanaka
+
+It supports 6P of [draft-03](draft-ietf-6tisch-6top-protocol-03) except for
+concurrent 6P transactions described in [Section
+4.3.3](https://tools.ietf.org/html/draft-ietf-6tisch-6top-protocol-03#section-4.3.3).
+
+## Code Structure
+
+6P is implemented in:
+- `core/net/mac/tsch/sixtop/sixtop.[ch]`: provides 6top APIs
+- `core/net/mac/tsch/sixtop/sixp.[ch]`: has charge of 6P other than packet manipulation
+- `core/net/mac/tsch/sixtop/sixp-packet[ch]`: provides 6P packet creater and parser
+
+## Configuration Paramters
+
+- `SIXTOP_CONF_MAX_SCHEDULE_FUNCTIONS`: the maximum number of SFs in 6top
+- `SIXTOP_CONF_6P_MAX_TRANSACTIONS`: the maximum number of transactions in process
+- `SIXTOP_CONF_6P_MAX_NEIGHBORS`: the maximum number of 6P neighbor states
+
+## Caveat
+
+At this moment, allocated 6P neighbor states never be deleted since 6P has no
+way to decide to delete them. 6P neighbor state consists of SeqNum to be used
```

Comment:
The Delete command doesn't help for this purpose. Even if a node has no cells with its peer as a result of the Delete command, it has to keep the neighbor state for the peer in order to:
- have a proper SeqNum value for the next request, whose value is obtained by **incrementing the previous SeqNum by exactly one**.
- keep GTX and GRX to avoid schedule generation inconsistency in the following transactions

To make matters worse, the node would keep the neighbor state even if the peer is gone...